pkg/storkctl/snapshot.go

```diff
@@ -220,3 +223,153 @@ func getSnapshotStatusAndTime(snap *snapv1.VolumeSnapshot) (string, string) {
+			if err != nil {
+				util.CheckErr(err)
+				return
+			}
+
+			msg := fmt.Sprintf("Snapshot %v restored successfully", restoreCRDName)
+			printMsg(msg, ioStreams.Out)
+		},
+	}
+	restoreSnapshotCommand.Flags().StringVarP(&snapName, "snapname", "n", "", "Snapshot name to be restored")
+	restoreSnapshotCommand.Flags().StringVarP(&snapNamespace, "namespace", "", "default", "Namespace of snapshot")
+	restoreSnapshotCommand.Flags().BoolVarP(&snapGroup, "is-groupsnapshot", "g", false, "True if snapshot is group, default false")
+	return restoreSnapshotCommand
+}
+
+func newGetVolumeSnapshotRestoreCommand(cmdFactory Factory, ioStreams genericclioptions.IOStreams) *cobra.Command {
+	restoreSnapshotCommand := &cobra.Command{
+		Use:     restoreSubCommand,
+		Aliases: []string{"restore"},
+		Short:   "Get snapshot restore status",
+		Run: func(c *cobra.Command, args []string) {
+			var snapRestoreList *storkv1.VolumeSnapshotRestoreList
+			var err error
+
+			namespaces, err := cmdFactory.GetAllNamespaces()
+			if err != nil {
+				util.CheckErr(err)
+				return
+			}
+
+			if len(args) > 0 {
+				snapRestoreList = new(storkv1.VolumeSnapshotRestoreList)
+				for _, restoreName := range args {
+					for _, ns := range namespaces {
+						snapRestore, err := k8s.Instance().GetVolumeSnapshotRestore(restoreName, ns)
+						if err != nil {
+							util.CheckErr(err)
+							return
+						}
+						snapRestoreList.Items = append(snapRestoreList.Items, *snapRestore)
+					}
+				}
+			} else {
+				var tempRestoreList storkv1.VolumeSnapshotRestoreList
+				for _, ns := range namespaces {
+					snapRestoreList, err = k8s.Instance().ListVolumeSnapshotRestore(ns)
+					if err != nil {
+						util.CheckErr(err)
+						return
+					}
+					tempRestoreList.Items = append(tempRestoreList.Items, snapRestoreList.Items...)
+				}
+				snapRestoreList = &tempRestoreList
+
+			}
+
+			if len(snapRestoreList.Items) == 0 {
+				handleEmptyList(ioStreams.Out)
+				return
+			}
+
+			if err := printObjects(c, snapRestoreList, cmdFactory, snapRestoreColumns, snapshotRestorePrinter, ioStreams.Out); err != nil {
+				util.CheckErr(err)
+				return
+			}
+		},
+	}
+
+	return restoreSnapshotCommand
+}
+
+func newDeleteVolumeSnapshotRestoreCommand(cmdFactory Factory, ioStreams genericclioptions.IOStreams) *cobra.Command {
+	restoreSnapshotCommand := &cobra.Command{
+		Use:     restoreSubCommand,
+		Aliases: []string{"restore"},
+		Short:   "Delete Volume snapshot restore",
+		Run: func(c *cobra.Command, args []string) {
+			if len(args) == 0 {
+				util.CheckErr(fmt.Errorf("at least one argument needs to be provided for volumesnapshotrestore name"))
+				return
+			}
+			name := args[0]
+			err := k8s.Instance().DeleteVolumeSnapshotRestore(name, cmdFactory.GetNamespace())
+			if err != nil {
+				util.CheckErr(err)
+				return
+			}
+
+			msg := fmt.Sprintf("Volume snapshot restore %v deleted successfully", name)
+			printMsg(msg, ioStreams.Out)
+		},
+	}
+
```

Comment:
Would be good to have a parameter here for the snapshot source name. If a name is given then delete all the restore objects that used that snapshot.

Reply:
should go in cloud snapshot PR, will add support In same PR